### PR TITLE
Display Terraform Version

### LINKED ISSUE #398: New version v1.1.0 does not work as expected
Hello, there is some issue with the latest version which has been released today - it determines wrongly Terraform version "09" while it was working fine until today. Please kindly fix the issue.

warrensbox/terraform-switcher info checking GitHub for latest tag
warrensbox/terraform-switcher info found version: 1.1.0 for v1.1.0/linux/amd64
warrensbox/terraform-switcher info installed /home/AzDevOps/.bin/tfswitch

Trying to read required terraform binary version from terraform files within /agent/_work/1/s/edited
TFVER=09

### PR DESCRIPTION
**Pull request by MatrixCrawler**
- add function to display the current version of terraform that was selected by tfswitch

Probably closes #398 and relates to #296

### CODE REVIEW

**Comment by warrensbox** [comment]
I would think that users would simply do `terraform -v` do get the latest version

**Comment by warrensbox** [comment]
Let me test this out

**Comment by yermulnik** [comment]
I actually am supporting what @warrensbox says, while I also can see the use case for this feature (e.g. when `tfswtich`-managed `terraform` **symlink** is called from the custom location which is not on the `PATH` or doesn't precede bin directory with another `terraform` **binary**).

Though I think this feature needs two improvements:
* Name of the parameter: I reckon it should be more specific, like e.g. `show-selected` to keep consistency with other similar parameters from https://github.com/warrensbox/terraform-switcher/blob/feature/which-flag/lib/param_parsing/parameters.go#L46C46-L48
* The logic should not rely upon the arbitrary file that can be changed from outside `tfswitch`. Since `tfswitch` installs TF binaries with version appended to the file name, I insist we resolve `terraform` symlink from the default bin dir (and respectively from `CustomBinaryPath` if defined by the user) to the real file and derive TF version from its name (was thinking that running resolved `terraform` binary should be much more reliable, though we re-link symlink w/o doing such, so it's superfluous).

**Comment by MatrixCrawler** [comment]
I will give this a thougt. Thanks for the input.

**Comment by MatrixCrawler** [comment]
I changed the approach in simply executing the binary, asking for the version and put out the data.

File: lib/param_parsing/parameters.go [hunk 1]
```diff
@@ -45,6 +46,7 @@ func GetParameters() Params {
 	getopt.BoolVarLong(&params.ShowLatestFlag, "show-latest", 'U', "Show latest stable version")
 	getopt.StringVarLong(&params.ShowLatestPre, "show-latest-pre", 'P', "Show latest pre-release implicit version. Ex: tfswitch --show-latest-pre 0.13 prints 0.13.0-rc1 (latest)")
 	getopt.StringVarLong(&params.ShowLatestStable, "show-latest-stable", 'S', "Show latest implicit version. Ex: tfswitch --show-latest-stable 0.13 prints 0.13.7 (latest)")
+	getopt.BoolVarLong(&params.ShowSelectedFlag, "which", 'w', "Displays the currently selected terraform version by tfswitch")
```
[the remark this answers is not in the document]
Comment (MatrixCrawler) [reply lib/param_parsing/parameters.go:49]: Forgot to change the flags....

**Review by yermulnik — COMMENTED** [review]
LGTM
If @warrensbox is Ok with the implementation (calling `terraform --version` from `installPath` from within `tfswitch`), then I'm all for it

File: lib/param_parsing/parameters.go [hunk 2]
```diff
@@ -45,6 +46,7 @@ func GetParameters() Params {
 	getopt.BoolVarLong(&params.ShowLatestFlag, "show-latest", 'U', "Show latest stable version")
 	getopt.StringVarLong(&params.ShowLatestPre, "show-latest-pre", 'P', "Show latest pre-release implicit version. Ex: tfswitch --show-latest-pre 0.13 prints 0.13.0-rc1 (latest)")
 	getopt.StringVarLong(&params.ShowLatestStable, "show-latest-stable", 'S', "Show latest implicit version. Ex: tfswitch --show-latest-stable 0.13 prints 0.13.7 (latest)")
+	getopt.BoolVarLong(&params.ShowSelectedFlag, "show-selected", 'w', "Display the currently selected terraform version by tfswitch")
```
Comment (crablab) [inline lib/param_parsing/parameters.go:49]:
```suggestion
	getopt.BoolVarLong(&params.ShowSelectedFlag, "show-selected", 'w', "Display the currently selected Terraform version by tfswitch")
```

Comment (crablab) [inline lib/param_parsing/parameters.go:49]: Where does the `w` option come from? 

Personally I would combine this with the `-v` option so it shows both the tfswitch version, and the currently selected Terraform version. I think this pattern is fairly common, where `-v` supplies versions for related components.

**Comment by warrensbox** [comment]
> LGTM If @warrensbox is Ok with the implementation (calling `terraform --version` from `installPath` from within `tfswitch`), then I'm all for it

@yermulnik @MatrixCrawler @crablab It really comes down to - "Do really need this feature". I feel like the user can simply do `terraform --version` to get the version. Also, it may be only one person's issue. If they can simply run `terraform --version`, or modify their process to get their version, we do not have to modify our code. 

For https://github.com/warrensbox/terraform-switcher/issues/296, the `tfswitch --show-latest` is actually working as expected. It should just show the latest binary that terraform has to offer.

**Comment by yermulnik** [comment]
> @yermulnik @MatrixCrawler @crablab It really comes down to - "Do really need this feature".

Citing myself =)
> [!NOTE]
> I actually am supporting what @warrensbox says, while I also can see the use case for this feature […]

I'm ok with declining this feature (the use case in #398 isn't really clear to me).

**Comment by crablab** [comment]
I'd agree with above - I don't think this is something we need to add. 
My comment above stands, in that if this is included I think it should be bundled with the existing `-v` option.

**Pull request closed without merging**